test/unit/lib/plugins/aws/provider.test.js

```diff
@@ -679,6 +605,92 @@ aws_secret_access_key = CUSTOMSECRET
     });
   });
 
+  describe('#getDeploymentPrefix()', () => {
+    it('should put all artifacts in namespaced folder', async () => {
+      const { cfTemplate } = await runServerless({
+        fixture: 'function',
+        command: 'package',
+      });
+      expect(cfTemplate).to.haveOwnProperty('Resources');
+      const resources = Object.values(cfTemplate.Resources);
+      const functions = resources.filter((r) => r.Type === 'AWS::Lambda::Function');
+      expect(functions.length).to.be.greaterThanOrEqual(1);
+      for (const f of functions) {
+        expect(f.Properties.Code.S3Key)
+          .to.be.a('string')
+          .and.satisfy((key) => key.startsWith('serverless/'));
+      }
+    });
+
+    it('should support custom namespaced folder', async () => {
+      const { cfTemplate } = await runServerless({
+        fixture: 'function',
+        command: 'package',
+        configExt: {
+          provider: {
+            deploymentPrefix: 'test-prefix',
+          },
+        },
+      });
+      expect(cfTemplate).to.haveOwnProperty('Resources');
+      const resources = Object.values(cfTemplate.Resources);
+      const functions = resources.filter((r) => r.Type === 'AWS::Lambda::Function');
+      expect(functions.length).to.be.greaterThanOrEqual(1);
+      for (const f of functions) {
+        expect(f.Properties.Code.S3Key)
+          .to.be.a('string')
+          .and.satisfy((key) => key.startsWith('test-prefix/'));
+      }
+    });
+  });
+
+  describe('#getAlbTargetGroupPrefix()', () => {
+    it('should support `provider.alb.targetGroupPrefix`', async () => {
+      const albId = '50dc6c495c0c9188';
+      const validBaseEventConfig = {
+        listenerArn: `arn:aws:elasticloadbalancing:us-east-1:123456789012:listener/app/my-load-balancer/${albId}/f2f7dc8efc522ab2`,
+        conditions: {
+          path: '/',
+        },
+      };
+      const { cfTemplate } = await runServerless({
+        fixture: 'function',
+        command: 'package',
+        configExt: {
+          provider: {
+            alb: {
+              targetGroupPrefix: 'a-prefix',
+            },
+          },
+          functions: {
+            fnTargetGroupName: {
+              handler: 'index.handler',
+              events: [
+                {
+                  alb: {
+                    ...validBaseEventConfig,
+                    priority: 1,
+                  },
+                },
+              ],
+            },
+          },
+        },
+      });
+      expect(cfTemplate).to.haveOwnProperty('Resources');
+      const resources = Object.values(cfTemplate.Resources);
```

Comment:
I think we can skip assignment to this temp variable which is used in only one place - let's go with `Object.values(cfTemplate.Resources).filter` in next line